rmgpy/solver/surface.pyx

```diff
@@ -384,17 +407,32 @@ cdef class SurfaceReactor(ReactionSystem):
 
         C = np.zeros_like(self.core_species_concentrations)
         V = self.V  # constant volume reactor
+        A = self.V * surface_volume_ratio_si  # area
+        total_sites = self.surface_site_density.value_si * A  # todo: double check units
 
+        current_surface_coverages = {}
         for j in range(num_core_species):
             if species_on_surface[j]:
                 C[j] = (N[j] / V) / surface_volume_ratio_si
+                surface_site_fraction = N[j] / total_sites
+                species = cov_dep_index_species[j]
+                current_surface_coverages[species] = surface_site_fraction
             else:
                 C[j] = N[j] / V
             #: surface species are in mol/m2, gas phase are in mol/m3
             core_species_concentrations[j] = C[j]
 
         for j in range(ir.shape[0]):
             k = kf[j]
+            if coverage_dependence is True:
+                if j in cov_dep:
+                    cov_dep_j = cov_dep[j]
+                    for species, parameters in cov_dep_j.items():
+                        if species in current_surface_coverages:
+                            theta = current_surface_coverages[species]
+                            if theta >= 1e-15:
+                                k *= 10. ** (parameters['a'].value_si * theta) * theta ** parameters['m'].value_si *\
+                                     np.exp(-1 * parameters['E'].value_si * theta / (constants.R * self.T.value_si))
```

Comment:
maybe we should make this a method in surface Arrhenius kinetics class instead of buried in the solver?

Reply:
bad idea

Reply:
It might be, but can you spare a few words to explain why?

Reply:
pros - increased visibility and testing, could use outside of context of solver
cons - harder to implement in solver because you would need to lug around the kinetics objects instead of passing them as arguments (attached to reactions) to initialize model function